File: AlexaDeviceFinder-Android/app/src/main/java/com/jamespfluger/alexadevicefinder/notifications/NotificationForge.java
package com.jamespfluger.alexadevicefinder.notifications;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.graphics.BitmapFactory;
import android.media.AudioAttributes;
import android.media.RingtoneManager;
import android.os.Build;

import androidx.annotation.RequiresApi;
import androidx.core.app.NotificationCompat;

import com.google.firebase.messaging.RemoteMessage;
import com.jamespfluger.alexadevicefinder.R;

public final class NotificationForge {
    private Context context;
    private NotificationManager notificationManager;
    private NotificationCompat.Builder notificationBuilder;
    private final String CHANNEL_ID = "CHANNEL_4096";

    public NotificationForge(Context context) {
        this.context = context;
        notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);

        notificationBuilder = new NotificationCompat.Builder(context, CHANNEL_ID);
        notificationBuilder.setChannelId(CHANNEL_ID)
                .setLargeIcon(BitmapFactory.decodeResource(context.getResources(), R.mipmap.ic_launcher))
                .setSmallIcon(R.mipmap.ic_notification)
                .setAutoCancel(true)
                .setOnlyAlertOnce(true)
                .setPriority(NotificationCompat.PRIORITY_MAX)
                .setCategory(NotificationCompat.CATEGORY_CALL);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && notificationManager.getNotificationChannel(CHANNEL_ID) == null) {
            createNotificationChannel();
        }
    }

    public void issueNotification(RemoteMessage remoteMessage) {
        notificationBuilder.setContentText(remoteMessage.getNotification().getTitle());
        notificationManager.notify((int) System.currentTimeMillis(), notificationBuilder.build());
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private void createNotificationChannel() {
        NotificationChannel channel = new NotificationChannel(
                CHANNEL_ID,
                context.getString(R.string.deviceAlertChannelName),
                NotificationManager.IMPORTANCE_HIGH);

        AudioAttributes ringtoneAttributes = new AudioAttributes.Builder()
                .setUsage(AudioAttributes.USAGE_NOTIFICATION_RINGTONE)
                .build();

        channel.setBypassDnd(true);
        channel.setSound(RingtoneManager.getActualDefaultRingtoneUri(this.context, RingtoneManager.TYPE_RINGTONE), ringtoneAttributes);

        notificationBuilder.setChannelId(CHANNEL_ID);
        notificationManager.createNotificationChannel(channel);
    }
}

File: MobileApp/DeviceFinder.Android/Notifications/NotificationForge.cs
using System;
using Android.App;
using Android.Content;
using Android.Graphics;
using Android.Media;
using Android.OS;
using Android.Util;
using AndroidX.Core.App;
using Firebase.Messaging;
using Xamarin.Essentials;

namespace DeviceFinder.Droid.Notifications
{
    public class NotificationForge
    {
        private readonly string channelId = "CHANNEL_4096";
        private readonly Context context;
        private readonly NotificationManager notificationManager;
        private readonly NotificationCompat.Builder notificationBuilder;

        public NotificationForge(Context context)
        {
            Log.Error("SAVEMEHELPME-NF", $"Context ctor: {context is not null}");
            Log.Error("SAVEMEHELPME-NF", $"Platform.AppContext: {Platform.AppContext is not null}");
            Log.Error("SAVEMEHELPME-NF", $"Platform.CurrentActvitity: {Platform.CurrentActivity is not null}");
            this.context = context;
            notificationManager = (NotificationManager)context.GetSystemService(Context.NotificationService);

            notificationBuilder = new NotificationCompat.Builder(context, channelId);
            notificationBuilder.SetChannelId(channelId)
                    .SetLargeIcon(BitmapFactory.DecodeResource(context.Resources, Resource.Mipmap.ic_launcher))
                    .SetSmallIcon(Resource.Mipmap.ic_notification)
                    .SetAutoCancel(true)
                    .SetOnlyAlertOnce(true)
                    .SetPriority(NotificationCompat.PriorityMax)
                    .SetCategory(NotificationCompat.CategoryCall);

            if (Build.VERSION.SdkInt >= BuildVersionCodes.O && notificationManager.GetNotificationChannel(channelId) == null)
            {
                CreateNotificationChannel();
            }
        }

        public void IssueNotification(RemoteMessage remoteMessage)
        {
            notificationBuilder.SetContentText(remoteMessage.GetNotification()?.Title ?? "Device Finder (null title)");
            notificationManager.Notify(DateTime.Today.Millisecond, notificationBuilder.Build());
        }

        private void CreateNotificationChannel()
        {
            NotificationChannel channel = new NotificationChannel(
                    channelId,
                    context.GetString(Resource.String.deviceAlertChannelName),
                    NotificationImportance.Max);

            AudioAttributes ringtoneAttributes = new AudioAttributes.Builder()
                    .SetUsage(AudioUsageKind.NotificationRingtone)
                    .Build();

            channel.SetBypassDnd(true);
            channel.SetSound(RingtoneManager.GetActualDefaultRingtoneUri(context, RingtoneType.Ringtone), ringtoneAttributes);

            notificationBuilder.SetChannelId(channelId);
            notificationManager.CreateNotificationChannel(channel);
        }
    }
}